src/SourceBuild/content/eng/pipelines/ci.yml

```diff
@@ -1,31 +1,29 @@
-# This is the main build definition (PR+CI) for dotnet/dotnet
+# This is the main build definition (CI) for dotnet/dotnet
+# This pipeline runs on a schedule every weekday at midnight (00:00 UTC) for "main" and when changes are pushed to "release/" and "internal/release/".
+
+schedules:
+- cron: '0 0 * * Mon-Fri'
+  displayName: Weekday midnight build
+  branches:
+    include:
+    - main
+  batch: true
 
 trigger:
   batch: true
   branches:
     include:
-    - main
     - release/*
     - internal/release/*
     exclude:
     - release/*.0.2xx
     - release/*.0.3xx
     - release/*.0.4xx
-    - internal/release/*.0.2xx
-    - internal/release/*.0.3xx
-    - internal/release/*.0.4xx
-
-pr:
-  branches:
-    include:
-    - main
-    - release/*
-    - internal/release/*
 
 stages:
-- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
-  - template: templates/stages/vmr-scan.yml
+- template: templates/stages/vmr-scan.yml

```

Comment:
Is this necessary given it is run by the light definition?

Reply:
I'm a bit confused on what you mean here. This yml defines the "full" dotnet-dotnet pipeline. As such, if I'm understanding your comment correctly, it won't be run by the light definition.

Reply:
[Removed running vmr-scan in heavy pipeline](https://github.com/dotnet/installer/pull/17370/files#diff-3bbdc81270db98c75c3c53e4ac2c8136f7dcd9c97fbe516b1e4d4737f5a45000L26)

Reply:
After refactoring the pipelines so that lite runs on main and all PRs, vmr-scan will only be triggered by main in that pipeline. As such, I added the vmr-scan stage back into ci.yml so that that stage can run for all branches except main.

https://github.com/dotnet/installer/pull/17370/commits/a0f1e49155c565fc1052096bdbeec7b0ccd9873b#diff-3bbdc81270db98c75c3c53e4ac2c8136f7dcd9c97fbe516b1e4d4737f5a45000R23